megamek/src/megamek/utils/MegaMekXmlUtil.java

```diff
@@ -141,22 +141,58 @@ public static void writeSimpleXmlTag(PrintWriter pw1, int indent, String name, S
         pw1.println(indentStr(indent) + "<" + name + ">" + escape(val) + "</" + name + ">");
     }
 
+    public static void writeSimpleXMLTag(PrintWriter pw1, int indent, String name, String... values) {
+        if (values.length > 0) {
+            pw1.println(indentStr(indent) + "<" + name + ">" + StringUtils.join(values, ',') + "</" + name + ">");
+        }
+    }
+
     public static void writeSimpleXmlTag(PrintWriter pw1, int indent, String name, int val) {
         pw1.println(indentStr(indent) + "<" + name + ">" + val + "</" + name + ">");
     }
 
+    public static void writeSimpleXMLTag(PrintWriter pw1, int indent, String name, int... values) {
+        if (values.length > 0) {
+            pw1.println(indentStr(indent) + "<" + name + ">" + StringUtils.join(values, ',') + "</" + name + ">");
+        }
+    }
+
     public static void writeSimpleXmlTag(PrintWriter pw1, int indent, String name, boolean val) {
         pw1.println(indentStr(indent) + "<" + name + ">" + val + "</" + name + ">");
     }
 
+    public static void writeSimpleXMLTag(PrintWriter pw1, int indent, String name, boolean... values) {
+        // cannot use StringUtils.join for a boolean array, so we are using this instead
+        StringBuilder csv = new StringBuilder();
```

Comment:
What about StringJoiner and `for (boolean value : values) joiner.add(String.valueOf(value));`

(Shorthand on phone)

Reply:
That's a solid idea. I started there and swapped it to a stream. I just took the code from Campaign Options and didn't consider reworking it.